core/jobs/batch_jobs/exp_validation_jobs.py

```diff
@@ -0,0 +1,261 @@
+# coding: utf-8
+#
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Validation jobs for explorations."""
+
+from __future__ import annotations
+
+from core.domain import exp_fetchers
+from core.domain import opportunity_services
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.transforms import job_result_transforms
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+
+MYPY = False
+if MYPY:  # pragma: no cover
+    from mypy_imports import exp_models
+
+(exp_models, ) = models.Registry.import_models([models.NAMES.exploration])
+
+
+class GetNumberOfInvalidExplorationsJob(base_jobs.JobBase):
+    """Job that retrieves the number of explorations that are invalid.
+    Here, invalid means that the explorations have the following properties:
+      a) The exploration is curated and has one or more states with
+          defined state classifier model id.
+      b) The exploration is curated and has non-empty param changes.
+      c) The exploration is curated and has non-empty param specs.
+      d) The exploration is curated and has non-empty training data.
```

Comment:
If you want you can combine last three into one, totally your call.
Thanks

Reply:
I just noticed that the 4th docstring was misleading. I have updated it now.